mapper/pojo-base/src/main/java/org/hibernate/search/mapper/pojo/massindexing/MassIndexerThreadLifecycleHooksConfigurer.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.mapper.pojo.massindexing;
+
+import org.hibernate.search.util.common.annotation.Incubating;
+
+/**
+ * Configurer that exposes thread lifecycle hooks that can be used to manipulate the execution environment, for example,
+ * to initialize {@link ThreadLocal ThreadLocals}. See interfaces extending {@link Context} to learn which threads will
+ * attempt to execute configured hooks.
+ */
+@Incubating
+public interface MassIndexerThreadLifecycleHooksConfigurer {
```

Comment:
That's a mouthful. Also, other configurers typically have a `configure` method, and other classes related to the `MassIndexer` typically use the `MassIndexing` prefix.

A few suggestions:

* `MassIndexingThreadScope` (inspired by JDK 20's "scoped values": https://openjdk.org/jeps/429)
* Simply `MassIndexingThreadEnvironment` or even `MassIndexingEnvironment`
* Something else?

I'd favor `MassIndexingEnvironment` since it's short and simple, WDYT?

Reply:
yup, that would be a better name 👍🏻 
I haven't changed the `supportsThreadLifecycleHooks()` as it seems more appropriate in the context of `PojoMassIndexingFailureHandledRunnable`. 
Added a row to the ref doc table now that the name should be more or less stable 😃 
Also, do not trust IDE to do refactoring correctly all the time. I've ended up with `MassIndexingMassIndexingMassIndexingEnvironmentIT` 😄